### ISSUE TITLE
Telegram notifications are not working

### ISSUE THREAD

**Issue by konikvranik** [roflcoopter/viseron#1255]
Hello, at first, thank you for a great piece of software! :-)
Now to the issue. I setup Telegram this way:

```yaml

telegram:
  telegram_bot_token: "xxxx"
  telegram_chat_ids:
    - 123456789
    - -1234567890123
  detection_label: person
  send_detection_thumbnail: true
  send_detection_video: true
  send_detection_message: true
  cameras:
    branka:
```

But Telegram notifications are not working for Telegram. Person detection is in the log and timeline event, and I receive a Gotify notification, but not Telegram's.

Telegram seems to be properly configured, because `/snapshot` works. Also `/ls`. But not notifications for a person.

**Comment by roflcoopter** [comment]
Your `telegram_chat_ids` look a bit funky, is that really how you have it configured or is it a mistake when you pasted it here?

**Comment by mgm-pvranik** [comment]
Of course I anonymized it by123.... The first one is the real ID of the bot, and the second one is the channel where the bot is admin (the channel starts with "-"). The connection between the bot and Viseron seems to be ok—commands work, but Viseron doesn't send me notifications about a person. But the event is logged in the timeline/events view.

**Comment by roflcoopter** [comment]
Alright, it works fine for me. I tried specifying two chat_ids and it works.

Could you enable debug logging for telegram and see if that gives any clues?
```yaml
logger:
  default_level: info
  logs:
    viseron.components.telegram: debug
    telegram: debug
```

**Comment by konikvranik** [comment]
I tried, but there was no message in the log from telegram. Just gotify.

**Comment by roflcoopter** [comment]
Which version of Viseron are you running? Can you try the `dev` version if you are not already using that?

**Comment by konikvranik** [comment]
Actually, I'm running the last dev one `Viseron / dev / 83ba965`. But I didn't try just `telegram: debug`. I added it to my config now. But to me it seems that the event is not reaching Telegram. Couldn't a preceding component eat the event?

**Comment by roflcoopter** [comment]
`telegram: debug` should tell us what the underlying telegram python lib is doing.

The event shouldnt be "eaten" by another component since Viseron employs pub/sub technology so there are many subscribers to the same events

**Comment by konikvranik** [comment]
Here it is - I forced person detection and received gotify message.

```
2026-02-18 17:42:06.159 [DEBUG   ] [telegram.ext.ExtBot] - Call to Bot API endpoint `getUpdates` finished with return value `[]`
2026-02-18 17:42:06.161 [DEBUG   ] [telegram.ext.ExtBot] - No new updates found.
2026-02-18 17:42:06.163 [DEBUG   ] [telegram.ext.ExtBot] - Calling Bot API endpoint `getUpdates` with parameters `{'timeout': 10, 'offset': 0}`
2026-02-18 17:42:07.781 [INFO    ] [viseron.domains] - Setup of domain face_recognition with identifier branka for component dlib took 4.8 seconds
2026-02-18 17:42:07.856 [INFO    ] [viseron.components.nvr.nvr.branka] - NVR for camera Branka initialized
2026-02-18 17:42:07.860 [INFO    ] [viseron.domains] - Setup of domain nvr with identifier branka for component nvr took 0.1 seconds
2026-02-18 17:42:07.900 [INFO    ] [viseron.core] - Viseron initialized in 23.4 seconds
2026-02-18 17:42:16.186 [DEBUG   ] [telegram.ext.ExtBot] - Call to Bot API endpoint `getUpdates` finished with return value `[]`
2026-02-18 17:42:16.187 [DEBUG   ] [telegram.ext.ExtBot] - No new updates found.
2026-02-18 17:42:16.188 [DEBUG   ] [telegram.ext.ExtBot] - Calling Bot API endpoint `getUpdates` with parameters `{'timeout': 10, 'offset': 0}`
2026-02-18 17:42:26.210 [DEBUG   ] [telegram.ext.ExtBot] - Call to Bot API endpoint `getUpdates` finished with return value `[]`
2026-02-18 17:42:26.212 [DEBUG   ] [telegram.ext.ExtBot] - No new updates found.
2026-02-18 17:42:26.213 [DEBUG   ] [telegram.ext.ExtBot] - Calling Bot API endpoint `getUpdates` with parameters `{'timeout': 10, 'offset': 0}`
2026-02-18 17:42:36.236 [DEBUG   ] [telegram.ext.ExtBot] - Call to Bot API endpoint `getUpdates` finished with return value `[]`
2026-02-18 17:42:36.238 [DEBUG   ] [telegram.ext.ExtBot] - No new updates found.
2026-02-18 17:42:36.239 [DEBUG   ] [telegram.ext.ExtBot] - Calling Bot API endpoint `getUpdates` with parameters `{'timeout': 10, 'offset': 0}`
2026-02-18 17:42:39.180 [INFO    ] [viseron.components.ffmpeg.recorder.branka] - Starting recorder
2026-02-18 17:42:39.237 [DEBUG   ] [viseron.components.gotify] - Created public image URL: https://viseron.suteren.net/api/v1/publicimage?token=a0fe06ba-6a33-4330-aecb-1dc3c301725f
2026-02-18 17:42:46.259 [DEBUG   ] [telegram.ext.ExtBot] - Call to Bot API endpoint `getUpdates` finished with return value `[]`
2026-02-18 17:42:46.260 [DEBUG   ] [telegram.ext.ExtBot] - No new updates found.
2026-02-18 17:42:46.261 [DEBUG   ] [telegram.ext.ExtBot] - Calling Bot API endpoint `getUpdates` with parameters `{'timeout': 10, 'offset': 0}`
2026-02-18 17:42:48.847 [INFO    ] [viseron.components.nvr.nvr.branka] - Stopping recording in: 10s
2026-02-18 17:42:50.403 [INFO    ] [viseron.components.nvr.nvr.branka] - Stopping recording in: 8s
2026-02-18 17:42:51.778 [INFO    ] [viseron.components.nvr.nvr.branka] - Stopping recording in: 7s
2026-02-18 17:42:52.854 [INFO    ] [viseron.components.nvr.nvr.branka] - Stopping recording in: 6s
2026-02-18 17:42:53.997 [INFO    ] [viseron.components.nvr.nvr.branka] - Stopping recording in: 5s
2026-02-18 17:42:54.878 [INFO    ] [viseron.components.nvr.nvr.branka] - Stopping recording in: 4s
2026-02-18 17:42:55.826 [INFO    ] [viseron.components.nvr.nvr.branka] - Stopping recording in: 3s
2026-02-18 17:42:56.283 [DEBUG   ] [telegram.ext.ExtBot] - Call to Bot API endpoint `getUpdates` finished with return value `[]`
2026-02-18 17:42:56.285 [DEBUG   ] [telegram.ext.ExtBot] - No new updates found.
2026-02-18 17:42:56.286 [DEBUG   ] [telegram.ext.ExtBot] - Calling Bot API endpoint `getUpdates` with parameters `{'timeout': 10, 'offset': 0}`
2026-02-18 17:42:57.143 [INFO    ] [viseron.components.nvr.nvr.branka] - Stopping recording in: 2s
2026-02-18 17:42:57.605 [INFO    ] [viseron.components.nvr.nvr.branka] - Stopping recording in: 1s
2026-02-18 17:42:58.831 [INFO    ] [viseron.components.nvr.nvr.branka] - Stopping recording in: 0s
2026-02-18 17:43:00.396 [INFO    ] [viseron.components.ffmpeg.recorder.branka] - Stopping recorder
2026-02-18 17:43:06.309 [DEBUG   ] [telegram.ext.ExtBot] - Call to Bot API endpoint `getUpdates` finished with return value `[]`
2026-02-18 17:43:06.310 [DEBUG   ] [telegram.ext.ExtBot] - No new updates found.
2026-02-18 17:43:06.312 [DEBUG   ] [telegram.ext.ExtBot] - Calling Bot API endpoint `getUpdates` with parameters `{'timeout': 10, 'offset': 0}`
2026-02-18 17:43:16.335 [DEBUG   ] [telegram.ext.ExtBot] - Call to Bot API endpoint `getUpdates` finished with return value `[]`
2026-02-18 17:43:16.336 [DEBUG   ] [telegram.ext.ExtBot] - No new updates found.
2026-02-18 17:43:16.337 [DEBUG   ] [telegram.ext.ExtBot] - Calling Bot API endpoint `getUpdates` with parameters `{'timeout': 10, 'offset': 0}`


2026-02-18 17:43:26.360 [DEBUG   ] [telegram.ext.ExtBot] - Call to Bot API endpoint `getUpdates` finished with return value `[]`
2026-02-18 17:43:26.361 [DEBUG   ] [telegram.ext.ExtBot] - No new updates found.
2026-02-18 17:43:26.363 [DEBUG   ] [telegram.ext.ExtBot] - Calling Bot API endpoint `getUpdates` with parameters `{'timeout': 10, 'offset': 0}`
2026-02-18 17:43:36.385 [DEBUG   ] [telegram.ext.ExtBot] - Call to Bot API endpoint `getUpdates` finished with return value `[]`
2026-02-18 17:43:36.386 [DEBUG   ] [telegram.ext.ExtBot] - No new updates found.
2026-02-18 17:43:36.388 [DEBUG   ] [telegram.ext.ExtBot] - Calling Bot API endpoint `getUpdates` with parameters `{'timeout': 10, 'offset': 0}`
```

**Comment by roflcoopter** [comment]
Strange, can you paste your full config so i can try to replicate it?
I dont have gotify setup but i cant see how that would interfere with Telegram

**Comment by konikvranik** [comment]
Here you are:
```
ffmpeg:
  camera:
    branka:
      name: Branka
      # host: localhost
      # port: 8554
      # path: /branka
      host: branka.home
      port: 554
      username: hass
      password: REDACTED
      path: /h264Preview_01_main
      protocol: rtsp
      input_args:
      - "-rtsp_transport"
      - "tcp"
      ffmpeg_recoverable_errors:
      - "error while decoding MB"
      - "invalid data found when processing input"
      substream:
        #  path: /branka_sub
        #  port: 8554
        path: /h264Preview_01_sub
        port: 554
        protocol: rtsp
        input_args:
        - "-rtsp_transport"
        - "tcp"
        fps: 3
      mjpeg_streams:
      recorder:
        lookback: 5
        idle_timeout: 10
        continuous_recording: true
        events:
          max_age:
            days: 90
        continuous:
          max_size:
            gb: 60

go2rtc:
  streams:
    branka:
    - rtsp://branka.home:554/h264Preview_01_main
    branka_sub:
    - rtsp://branka.home:554/h264Preview_01_sub

mog2:
  motion_detector:
    cameras:
      branka:
        detect_shadows: true
        trigger_event_recording: false
        #learning_rate: 0.01

yolo:
  object_detector:
    model_path: /detectors/models/yolo/yolov8n.pt
    cameras:
      branka:
        fps: 1
        scan_on_motion_only: true
        log_all_objects: true
        labels:
        - label: person
        - label: bicycle
        - label: car
          require_motion: true
        - label: motorcycle
          require_motion: true
        - label: airplane
          require_motion: true
        - label: bus
          require_motion: true
        - label: truck
          require_motion: true
        - label: bird
        - label: cat
        - label: dog
        - label: horse
        - label: sheep
        - label: cow
        - label: elephant
        - label: bear
        - label: zebra
        - label: giraffe
        - label: backpack
        - label: umbrella
        - label: handbag
        - label: tie
        - label: suitcase
        - label: frisbee
        - label: skis
        - label: snowboard
        - label: sports ball
        - label: kite
        - label: baseball bat
        - label: baseball glove
        - label: skateboard
        - label: surfboard
        - label: tennis racket
        - label: bottle
        - label: wine glass
        - label: cup
        - label: fork
        - label: knife
        - label: spoon
        - label: bowl
        - label: banana
        - label: apple
        - label: sandwich
        - label: orange
        - label: broccoli
        - label: carrot
        - label: hot dog
        - label: pizza
        - label: donut
        - label: cake
        - label: chair
        - label: couch
        - label: potted plant
        - label: bed
        - label: dining table
        - label: toilet
        - label: tv
        - label: laptop
        - label: mouse
        - label: remote
        - label: keyboard
        - label: cell phone
        - label: microwave
        - label: oven
        - label: toaster
        - label: sink
        - label: refrigerator
        - label: book
        - label: clock
        - label: vase
        - label: scissors
        - label: teddy bear
        - label: hair drier
        - label: toothbrush
dlib:
  face_recognition:
    model: hog
    expire_after: 10
    labels:
    - person
    face_recognition_path: /config/face_recognition/faces
    save_unknown_faces: true
    save_faces: true
    cameras:
      branka:

codeprojectai:
  host: codeproject-ai.ai.svc.cluster.local
  port: 32168
  timeout: 30
  license_plate_recognition:
    cameras:
      branka:
    labels:
    - vehicle
    - car
    - truck
    - motorcycle
    - motorbike
    known_plates:
    - 4SU9442
    - 3SY0941
    expire_after: 10

gotify:
  gotify_url: "https://gotify.home" # URL to your Gotify server
  gotify_token: "REDACTED" # Application token from Gotify
  priority: 5 # Priority of the notifications (1-10)
  detection_label: "person,cat" # Labels of objects to send notifications for
  send_thumbnail: true # Send a thumbnail of the detected object
  use_public_url: true # Generate a temp public URL with a unique token for the image. Require public_base_url in webserver conf
  image_max_size: 800 # Maximum width/height in pixels for thumbnails (default: 800). For 4K cameras, use 1920 or higher
  image_quality: 95 # JPEGAccess denied for user 83364792. quality for images (1-100, default: 95). Higher = better quality but larger files
  cameras:
    branka:

telegram:
  telegram_bot_token: "REDACTED"
  telegram_chat_ids:
  - REDACTED
  - -REDACTED
  detection_label: person
  send_detection_thumbnail: true
  send_detection_video: true
  send_detection_message: true
  cameras:
    branka:

nvr:
  branka:

webserver:
  public_base_url: https://viseron.home
  auth:
    enabled: true

mqtt:
  broker: mqtt.home
  port: 1883
  client_id: viseron
  home_assistant:

logger:
  default_level: info
  logs:
    viseron.components.telegram: debug
    viseron.components.gotify: debug
#    viseron.components.mog2: debug
#    viseron.components.yolo: debug
    telegram: debug
```

**Comment by roflcoopter** [comment]
And if you disable `gotify` you still dont get `telegram` notifications? Just to rule that out.

**Comment by mgm-pvranik** [comment]
No messages even with gotify disable,

**Comment by roflcoopter** [comment]
Ahh now i get it. You need to add `create_event_clip: true` under the `recorder` block, since the telegram component listens to that event before sending notifications (so that it can send the `mp4` file).

Not ideal i know, would be better if the snapshots and message was sent instantly and only the video portion would depend on `create_event_clip`.

Will fix that once #1205 is done since that PR has some changes to the Telegram component

**Comment by konikvranik** [comment]
OIC, so if I'll set `send_detection_video: false`, it should help too, right? Thank you for the hint, I'll try it.

**Comment by roflcoopter** [comment]
No it wont sadly, the component listens for the event that triggers when the file created by `create_event_clip` is complete, which is not correct imo.
https://github.com/roflcoopter/viseron/blob/78c00afd018b37e16ab3794a035d4abf60c6b67a/viseron/components/telegram/__init__.py#L201-L205

**Comment by konikvranik** [comment]
OK, so with `create_event_clip: true` I'm getting notifications on Telegram. But another issue is that even if I have `detection_label: person` set, I'm also receiving events initiated by the `car` label with no person detected.
